src/signature.c

```diff
@@ -0,0 +1,271 @@
+        HANDLE_ERROR("Failed to create maps path");
+    }
+
+    FILE* maps_file = fopen(maps_path, "r");
+    if (!maps_file) {
+        HANDLE_ERROR("Failed to open maps file");
+    }
+
+    MemoryRegion* regions = NULL;
+    int capacity = 0;
+    *count = 0;
+
+    char line[256];
+    while (fgets(line, sizeof(line), maps_file)) {
+        if (*count >= capacity) {
+            capacity = capacity == 0 ? 10 : capacity * 2;
+            MemoryRegion* temp = realloc(regions, capacity * sizeof(MemoryRegion));
+            if (!temp) {
+                free(regions);
+                fclose(maps_file);
+                HANDLE_ERROR("Failed to allocate memory for regions");
+            }
+            regions = temp;
+        }
+
+        uintptr_t start, end;
+        if (sscanf(line, "%" SCNxPTR "-%" SCNxPTR, &start, &end) != 2) {
+            continue; // Skip lines that don't match the expected format
+        }
+        regions[*count].start = start;
+        regions[*count].end = end;
+        (*count)++;
+    }
+
+    fclose(maps_file);
+    return regions;
+}
+
+/**
+ * Matches a pattern with an array of bytes.
+ *
+ * @param data The data to compare the pattern against.
+ * @param pattern The pattern to test for.
+ * @param pattern_size The length of the pattern.
+ *
+ * @return True if the pattern matches the data, false otherwise
+ */
+bool match_pattern(const uint8_t* data, const uint16_t* pattern, size_t pattern_size)
+{
+    for (size_t i = 0; i < pattern_size; ++i) {
+        uint8_t byte = pattern[i] & 0xFF;
+        bool ignore = (pattern[i] >> 8) & 0x1;
+        if (!ignore && data[i] != byte) {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * Converts an IDA-like signature into a pattern to be used in LibreSplit.
+ * Supports the '??' string to ignore certain bytes in the comparison.
+ *
+ * @param signature A string containing the signature to convert.
+ * @param pattern_size A pointer onto where to save the size of the pattern.
+ *
+ * @return A pattern to be used with the LibreSplit signature scan functions.
+ */
+uint16_t* convert_signature(const char* signature, size_t* pattern_size)
+{
+    char* signature_copy = strdup(signature);
+    if (!signature_copy) {
+        return NULL;
+    }
+
+    char* token = strtok(signature_copy, " ");
+    size_t size = 0;
+    size_t capacity = 10;
+    uint16_t* pattern = (uint16_t*)malloc(capacity * sizeof(uint16_t));
+    if (!pattern) {
+        free(signature_copy);
+        return NULL;
+    }
+
+    while (token != NULL) {
+        if (size >= capacity) {
+            capacity *= 2;
+            uint16_t* temp = (uint16_t*)realloc(pattern, capacity * sizeof(uint16_t));
+            if (!temp) {
+                free(pattern);
+                free(signature_copy);
+                return NULL;
+            }
+            pattern = temp;
+        }
+
+        if (strcmp(token, "??") == 0) {
```

Comment:
Also according to VVVVVV splitter, it should also handle 0? and ?0 the same as ??

https://raw.githubusercontent.com/tzann/Autosplitters/master/LiveSplit.VVVVVV.asl
> // Note: Masks always mask the entire byte, so 0? and ?0 are the same as ??